monitoring/monitorlib/auth.py

```diff
@@ -254,33 +253,57 @@ def issue_token(self, intended_audience: str, scopes: List[str]) -> str:
 
 
 class ClientIdClientSecret(AuthAdapter):
-  """Auth adapter that gets JWTs using a client ID and client secret."""
+  """Auth adapter that gets JWTs using a client ID and client secret. By default, this will send the request as JSON, you can use send_request_as_data flag to send the request as form data. """
 
-  def __init__(self, token_endpoint: str, client_id: str, client_secret: str):
+  def __init__(self, token_endpoint: str, client_id: str, client_secret: str, send_request_as_data :bool=False):
     super().__init__()
 
     self._oauth_token_endpoint = token_endpoint
     self._client_id = client_id
     self._client_secret = client_secret
-
+    self.send_request_as_data = send_request_as_data
```

Comment:
nit: Python convention is for private class fields to be prefixed with a single underscore (see fields above this one).  Tools in a shared/reusable library generally carry a higher bar for documentation, so for fields that consumers of class instances will not need to use, a single underscore often removes the need to document the field by marking it as not for use by class instance consumers.  Otherwise, the field should usually be documented.

Reply:
Thanks 🙏🏼 , yes I have now fixed it.